src/api/types.ts

```diff
@@ -1413,7 +1413,7 @@ export type SearchHighlighterType = 'plain' | 'fvh' | 'unified' | string
 
 export interface SearchHit<TDocument = unknown> {
   _index: IndexName
-  _id: Id
+  _id?: Id
```

Comment:
How can I figure out why this can be `undefined` now? I can't seem to notice anything relevant in changelogs.

Reply:
Type definitions are generated from [the specification](https://github.com/elastic/elasticsearch-specification). In this particular case, it looks like this change happened in https://github.com/elastic/elasticsearch-specification/pull/2592.

Reply:
Thank you! This clears it up.

```
 * @es_quirk '_id' is not available when using 'stored_fields: _none_'
 * on a search request. Otherwise the field is always present on hits
 ```